ouroboros-network/src/Ouroboros/Network/PeerSelection/State/EstablishedPeers.hs

```diff
@@ -162,13 +163,16 @@ member peeraddr = Map.member peeraddr . allPeers
 insert :: Ord peeraddr
        => peeraddr
        -> peerconn
-       -> Time
+       -> Maybe Time
        -> EstablishedPeers peeraddr peerconn
        -> EstablishedPeers peeraddr peerconn
-insert peeraddr peerconn peerShareAt ep@EstablishedPeers { allPeers } =
+insert peeraddr peerconn peerShareAt_m ep@EstablishedPeers { allPeers } =
    -- Ask newly established peers for its peers after the specified delay.
-   setPeerShareTime (Set.singleton peeraddr) peerShareAt $
-     ep { allPeers = Map.insert peeraddr peerconn allPeers }
+   let ep' = ep { allPeers = Map.insert peeraddr peerconn allPeers } in
+   case peerShareAt_m of
+        Nothing          -> ep'
+        Just peerShareAt ->
+          setPeerShareTime (Set.singleton peeraddr) peerShareAt $ ep'
```

Comment:
I think this is a bit cleaner (at the expense of code duplication, which could be eliminated with a recursive call):
```suggestion
insert peeraddr peerconn Nothing ep@EstablishedPeers { allPeers } =
    ep { allPeers = Map.insert peeraddr peerconn allPeers }   
insert peeraddr peerconn (Just peerShareAt) ep@EstablishedPeers { allPeers } =
    -- Ask newly established peers for its peers after the specified delay.
    setPeerShareTime (Set.singleton peeraddr) $
    ep { allPeers = Map.insert peeraddr peerconn allPeers } 
```